### PR TITLE
[DOCS] Expands the documentation of Node Query Cache

### LINKED ISSUE #47955: Expand on Node Query Cache section of docs
**Describe the feature**:

I've found a lot of interesting pieces of information about the Node Query Cache that'd be good to capture in the docs (provided they're accurate and still applicable in 7.x). Things such as:

* Term queries are not eligible for caching, even in the filter context [1]
* Caching is done on a per segment basis, but only under certain conditions (e.g a segment of >=10k documents + >=3% of total shard size [2])
* Due to the per segment basis, merging can invalidate a cached query [3]
* There is a finite number of queries that will be cached in the available heap space [2]
* To determine if a query is eligible for caching, we maintain a history to track occurrences and this history as a finite number of queries it will track [4]

[1] https://github.com/elastic/elasticsearch/pull/27190
[2] https://discuss.elastic.co/t/query-cache-alternatives-to-term-query/180188/4
[3] https://discuss.elastic.co/t/shard-request-cache-vs-node-query-cache/91046/2
[4] https://lucene.apache.org/core/8_2_0/core/org/apache/lucene/search/UsageTrackingQueryCachingPolicy.html

### PR DESCRIPTION
**Pull request by szabosteve**
This PR adds further information to the [Node Query Cache documentation](https://www.elastic.co/guide/en/elasticsearch/reference/master/query-cache.html).

Related issue that this PR closes https://github.com/elastic/elasticsearch/issues/47955

Preview: http://elasticsearch_51105.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-cache.html

### CODE REVIEW

File: docs/reference/modules/indices/query_cache.asciidoc [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
-The query cache is responsible for caching the results of queries.
-There is one queries cache per node that is shared by all shards.
-The cache implements an LRU eviction policy: when a cache becomes full, the
-least recently used data is evicted to make way for new data.
-It is not possible to look at the contents being cached.
+The query cache is responsible for caching the results of queries. There is one 
+queries cache per node that is shared by all shards. The cache implements an LRU 
+eviction policy: when a cache becomes full, the least recently used data is 
+evicted to make way for new data. It is not possible to look at the contents 
+being cached.
```
Comment (debadair) [inline docs/reference/modules/indices/query_cache.asciidoc]: I'd take the opportunity to tighten this up a bit. 

The results of queries used in the filter context are cached in the node query cache for fast lookup.
The cache uses an LRU eviction policy: when the cache is full, the least recently used query results are evicted to make way for new data. You cannot inspect the contents of the query cache.

File: docs/reference/modules/indices/query_cache.asciidoc [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
 
-The query cache only caches queries which are being used in a filter context.
+The query cache only caches queries which are being used in a filter context. 
+However, term queries are not eligible for caching, not even in the filter 
+context.
```
Comment (debadair) [inline docs/reference/modules/indices/query_cache.asciidoc]: If you mention that this cache is for queries used in a filter context above, you can probably just focus on the exceptions here:

Term queries and queries used outside of a filter context are not eligible for caching.

File: docs/reference/modules/indices/query_cache.asciidoc [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
+
+It caches up to 10000 queries in up to 10% of the total heap space by default.
```
Comment (debadair) [inline docs/reference/modules/indices/query_cache.asciidoc]:
```suggestion
By default, the cache holds a maximum of 10000 queries in up to 10% of the total heap space. 
```

File: docs/reference/modules/indices/query_cache.asciidoc [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
 
-The query cache only caches queries which are being used in a filter context.
+The query cache only caches queries which are being used in a filter context. 
+However, term queries are not eligible for caching, not even in the filter 
+context.
+
+It caches up to 10000 queries in up to 10% of the total heap space by default. 
+To determine if a query is eligible for caching, a history of queries is 
+maintained to track occurrences.
+
```
Comment (debadair) [inline docs/reference/modules/indices/query_cache.asciidoc]: I'd probably switch this to the active voice:

To determine if a query is eligible for caching, Elasticsearch maintains a query history to track occurrences.

File: docs/reference/modules/indices/query_cache.asciidoc [hunk 5]
```diff
@@ -1,27 +1,35 @@
...
+Caching is done on a per segment basis if a segment contains at least 10000 
+documents and the segment has at least 3% of the total documents of a shard. Due 
+to the per segment basis, segment merging can invalidate a cached query.
```
Comment (debadair) [inline docs/reference/modules/indices/query_cache.asciidoc]: Maybe:

Because caching is per segment, merging segments can invalidate cached queries.

**Review by debadair — APPROVED** [review]
Thanks for taking this one on! Left some minor editorial comments--use if they're helpful.

**Comment by szabosteve** [comment]
@debadair Thank you very much for your comments, they are very helpful. I modified the text accordingly.

**Comment by szabosteve** [comment]
**Backport commits:**
7.x: 424b4ed
7.6: 09c8bcf
7.5: 48ab639

**Pull request merged**